src/frontend.rs

```diff
@@ -0,0 +1,276 @@
+#![allow(non_snake_case)]
+
+use dioxus::prelude::*;
+use std::fmt::{self, Display, Formatter};
+use std::num::ParseFloatError;
+use std::str::FromStr;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsCast;
+use wasm_bindgen_futures::spawn_local;
+use wasm_bindgen_futures::JsFuture;
+use web_sys::console;
+use web_sys::Request;
+use web_sys::RequestInit;
+use web_sys::RequestMode;
+use web_sys::Response;
+use web_sys::WebSocket;
+
+#[wasm_bindgen(start)]
+pub fn run_app() {
+    launch(App);
+}
+
+#[derive(PartialEq, Props, Clone, Debug)]
+struct Scores {
+    o: f32,
+    c: f32,
+    e: f32,
+    a: f32,
+    n: f32,
+}
+
+impl Display for Scores {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{},{},{},{},{}", self.o, self.c, self.e, self.a, self.n)
+    }
+}
+
+impl FromStr for Scores {
+    type Err = ParseFloatError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let values: Vec<&str> = s.split(',').collect();
+        if values.len() != 5 {}
+
+        let o = values[0].parse()?;
+        let c = values[1].parse()?;
+        let e = values[2].parse()?;
+        let a = values[3].parse()?;
+        let n = values[4].parse()?;
+
+        Ok(Self { o, c, e, a, n })
+    }
+}
+
+impl Scores {
+    fn from_form(form: &FormData) -> Option<Self> {
+        let data = form.values();
+
+        let o: f32 = data.get("o")?.as_value().parse().ok()?;
+        let c: f32 = data.get("c")?.as_value().parse().ok()?;
+        let e: f32 = data.get("e")?.as_value().parse().ok()?;
+        let a: f32 = data.get("a")?.as_value().parse().ok()?;
+        let n: f32 = data.get("n")?.as_value().parse().ok()?;
+
+        if o < 0. || o > 100. {
+            return None;
+        }
+        if c < 0. || c > 100. {
+            return None;
+        }
+        if e < 0. || e > 100. {
+            return None;
+        }
+        if a < 0. || a > 100. {
+            return None;
+        }
+        if n < 0. || n > 100. {
+            return None;
+        }
+
+        Some(Self { o, c, e, a, n })
+    }
+}
```

Comment:
I don't know if Scores should be in the frontend file, couldn't this logic be used in the server in the future?

Reply:
way ahead of you, check the next PR!